src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -617,7 +617,12 @@ mongoc_topology_destroy (mongoc_topology_t *topology)
       bson_mutex_destroy (&topology->apm_mutex);
       mongoc_cond_destroy (&topology->srv_polling_cond);
    }
-   _mongoc_topology_description_monitor_closed (&topology->description);
+
+   if (topology->valid) {
```

Comment:
+1 I'm assuming that this is okay via spec.

Reply:
[The SDAM monitoring spec](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring-monitoring.rst#specification) defines the `TopologyOpening` event as:

> When a topology description is initialized - this MUST be the first SDAM event fired

I interpret that to mean it is OK to omit a `TopologyOpening` event if the topology fails to initialize. That is the current behavior. And I think it follows that a `TopologyClosed` event must not be omitted either.